index.js

```diff
@@ -166,15 +170,15 @@ module.exports = class ServeDrive extends ReadyResource {
     }
 
     try {
-      if (drive !== null) await this.releaseDrive(id)
+      if (drive !== null) await this.releaseDrive(id, drive)
     } catch (e) {
       safetyCatch(e)
       // can technically can overwrite the prev error, but we are ok with that as these
       // are for simple reporting anyway and this is the important one.
       error = e
     }
 
-    if (error === null) return
+    if (this.closing || error === null) return
```

Comment:
Is this a separate bugfix? Not sure what this would mean for the user if the request had not yet sent a response

Reply:
same reason as above, the user closed the instance - all connections are dead. we should stop. can happen during one of the prev awaits